var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -60,15 +60,21 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("intel-oneapi-dpl", when="+oneapi")
 
     conflicts("%gcc@:5.2.9")
-    conflicts("+rocm", when="@:1.1.1")
+
+    # ROCm should work with this custom branch
+    with when(not "@glu_experimental"):
+        conflicts("+rocm", when="@:1.1.1")
```

Comment:
It's surprising that you need to do this. I thought this would apply only to versioned variants and not named ones? At least, that was my intention with this code, because both `develop` and `master` should hopefully not be caught in this conflict.

Also, with ROCm 5.2.0, there was some error in rocrand's CMake setup, I am not sure if you faced them on your system. See: https://github.com/RadeonOpenCompute/rocm-cmake/pull/107 (we need to either use this fix or also give `-Drocrand_DIR=.../lib/cmake/rocrand` and `-Dhiprand_DIR=.../lib/cmake/hiprand` when building Ginkgo.

Reply:
I don't seem to have run into this issue. I think I ran into this with 4.5.0, but I wasn't quite aware if this was just an issue with our installation, or otherwise. Perhpas I can get the 4.5.0 installation to work with this knowledge!

I specified my rocrand explicitly as an external in my `spack.yaml` for the environment I was using, and it's possible that on the ORNL system (Crusher) they have already implemented the CMake fix...